ocaml/xapi/xapi_vgpu.ml

```diff
@@ -42,19 +62,27 @@ let create' ~__context  ~vM ~gPU_group ~device ~other_config ~_type ~powerstate_
                     (Api_errors.invalid_value, ["type"; Ref.string_of _type]))
     end
   in
+  (* during multiple vgpus creation:
+     1. Underlying vgpu_type should support multiple
+     2. _type must be listed on the all vgpu_type's compatible lists*)
+  let existing = Db.VM.get_VGPUs ~__context ~self:vM in
+  if existing <> [] then begin
+    let types = List.map (fun vgpu -> Db.VGPU.get_type ~__context ~self:vgpu) existing in
+    let is_in_compatible_lists _type vgpu_type =
+      let compatible_lists = Db.VGPU_type.get_compatible_types_in_vm ~__context ~self:vgpu_type in
+      List.mem _type compatible_lists
```

Comment:
It is a bit unusual to use `_type` (with a an underscore) for a parameter that is actually uses. In particular, when this is a named parameter. We can leave this for now.

Reply:
I think that this was done because `type` is a keyword. In fact, the datamodel code generator may have added this prefix.